### DISCUSSION TITLE
Is this the only way to get valid units?

### DISCUSSION THREAD

**Discussion by AFZL210** [josdejong/mathjs#3181] [Q&A] [answered]
Hi I asked this question related to earlier on how to to get all the available units and just wanted to confirm that the solution in this https://github.com/josdejong/mathjs/discussions/2805 is the only way to get valid units list?

Thanks!

**Comment by dvd101x** [comment, marked as answer]
Hi, I wouldn't say it's the only way. There might be many ways, but I would recommend exploring `math.Unit.UNITS` to find a way to match your specific needs.

I personally think the answer proposed at https://github.com/josdejong/mathjs/discussions/3178#discussioncomment-8870819 is very complete. Maybe you don't need all prefixes so you can skip some steps.

You could find similar solutions in various places in this project. Which might be useful to your needs.
- https://github.com/josdejong/mathjs/blob/02f195f70003a588cc5d4f0aad910aff2a2a57b7/bin/cli.js#L123-L152
- https://github.com/josdejong/mathjs/blob/9e86a6cec611582dd9bed075c112e45cb92cd7fc/js/commandline.js#L224-L260
- https://github.com/josdejong/mathjs/blob/02f195f70003a588cc5d4f0aad910aff2a2a57b7/examples/code%20editor/mathjs-lang.js#L199-L231

**Reply by AFZL210** [reply]
Thank you @dvd101x
